pkg/engine/shield/add.go

```diff
@@ -4,20 +4,46 @@ import (
 	"github.com/simimpact/srsim/pkg/engine/event"
 	"github.com/simimpact/srsim/pkg/engine/info"
 	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
 )
 
 func (mgr *Manager) AddShield(id key.Shield, shield info.Shield) {
 	// TODO: LOGIC FOR ADDING A SHIELD
 	// 1. Check if the target already has this shield, if so remove old (mgr.RemoveShield)
+	if mgr.HasShield(shield.Target, id) {
+		mgr.RemoveShield(id, shield.Target)
+	}
+
 	// 2. Get the stats for the source and target
-	// 3. Compute shield HP/create ShieldInstance given the add paramsm
+	// 3. Compute shield HP/create ShieldInstance given the add params
+	shieldHP := 0.0
+
+	for k, v := range shield.BaseShield {
+		switch k {
+		case model.ShieldFormula_SHIELD_BY_SHIELDER_ATK:
+			shieldHP += v * mgr.attr.Stats(shield.Source).ATK()
+		case model.ShieldFormula_SHIELD_BY_SHIELDER_DEF:
+			shieldHP += v * mgr.attr.Stats(shield.Source).DEF()
+		case model.ShieldFormula_SHIELD_BY_SHIELDER_MAX_HP:
+			shieldHP += v * mgr.attr.Stats(shield.Source).HP()
+		case model.ShieldFormula_SHIELD_BY_TARGET_MAX_HP:
+			shieldHP += v * mgr.attr.Stats(shield.Target).HP()
+		case model.ShieldFormula_SHIELD_BY_SHIELDER_TOTAL_SHIELD:
+			shieldHP += shield.ShieldValue
+		}
+	}
```

Comment:
* nit: Make the `Stats` calls before this loop. We want to minimize the amount of times we need to call `Stats` as much as possible (this will call `Stats` multiple times for the same target in the case of multiple formulas used)
* for `SHIELD_BY_SHIELDER_TOTAL_SHIELD` I believe this formula should be v times the current maximum shield HP attached to the shielder. Think of it like "duplicating" the shield from shielder to target. Should make helper to get the max/total shield HP